### PR TITLE
Update bpmn-coverage.md

### PR DESCRIPTION
**Pull request by aleksander-dytko**
We support Data Object and Data Store in Modeler + for execution. This needs to be reflected in bpmn-coverage

## When should this change go live?

- [ ] This is a bug fix, security concern, or something that needs **urgent release support**.
- [x] This is already available but undocumented and should be released within a week.
- [ ] This on a **specific schedule** and the assignee will coordinate a release with the DevEx team. (apply `hold` label or convert to draft PR)
- [ ] This is part of a scheduled alpha or minor. (apply alpha or minor label)
- [ ] There is **no urgency** with this change and can be released at any time.

## PR Checklist

- [ ] My changes are for an **already released minor** and are in `/versioned_docs` directory.
- [ ] My changes are for the **next minor** and are in `/docs` directory (aka `/next/`).

- [x] My changes require an [Engineering review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned the Engineering DRI or delegate.
- [x] My changes require a [technical writer review](https://github.com/camunda/camunda-docs/blob/main/howtos/documentation-guidelines.md#review-process), and I've assigned @camunda/tech-writers as a reviewer.

### CODE REVIEW

**Review by korthout — CHANGES_REQUESTED** [review]
While we allow processes with DataObject and DataStore, we don't really support these BPMN concepts. We simply don't use them in the execution.

- https://github.com/camunda/camunda/pull/10845

I'm not confident in saying that we provide coverage for these concepts. I believe this would require using `DataInput` and `DataOutput` for `Activities` like `Service Tasks`. Instead, we just don't break when BPMN models contain `DataObject` and `DataStore`, but they are not actively used in the execution of the model.

IMO, we should discuss this with other BPMN experts before merging.

**Comment by aleksander-dytko** [comment]
@korthout thanks for sharing your opinion. 

@saig0 @nikku could you support us here to define what would be needed so that we can claim we support Data store and data object elements?

**Comment by nikku** [comment]
> @saig0 @nikku could you support us here to define what would be needed so that we can claim we support Data store and data object elements?

TLDR: Data stores, input and output have always been _modeling only_ from Camunda's perspective, in C8 and C7. We don't support any of the "data features" beyond modeling (for documentation purposes) that the BPMN standard defines.

----

BPMN implements data on two layers:

* Visual, i.e. data stores and data objects, documentation only
* For execution (+visual) using IO mappings (data input, and data output)

The later one is a direct competition to our own data mappings (both C7 and C8).

An answer to our users could be to mark them as supported (*) with the disclaimer that this includes "modeling only/for documentation" and that we do not support IO-mappings as defined in the BPMN standard.

**Review by christinaausley — APPROVED** [review]

**Review by christinaausley — COMMENTED** [review]
Let me know when you're ready for final review @aleksander-dytko 👍

**Comment by akeller** [comment]
@aleksander-dytko - nudging you on this docs PR.

**Comment by christinaausley** [comment]
@aleksander-dytko Ready for final review here? 😄

**Comment by christinaausley** [comment]
Followed up with Aleksander on Slack and I'm going to help refactor this PR to include Nico’s [suggestion](https://github.com/camunda/camunda-docs/pull/4155#issuecomment-2288983434) (disclaimer for modeling only).

**Comment by christinaausley** [comment]
@aleksander-dytko I've incorporated a disclaimer for `DataObject` and `DataStore` -- if this looks good to you, I will backport.

File: docs/components/modeler/bpmn/bpmn-coverage.md [hunk 1]
```diff
@@ -140,15 +140,19 @@ import CompensationSvg from './assets/bpmn-symbols/compensation.svg'
 
 ## Data
 
+:::note
+`DataObject` and `DataStore` are supported by Camunda for modeling purposes only. Camunda does not support any additional data features the BPMN standard defines.
```
Comment (nikku) [inline docs/components/modeler/bpmn/bpmn-coverage.md:144]: We could also frame this as "we don't support BPMN standard IO mappings (but of course allow you to map IO (in our very simple, CAmunda style) [link to our documentation for that]". This may read as "We do not support data features".

Comment (christinaausley) [reply docs/components/modeler/bpmn/bpmn-coverage.md:144]: How about "DataObject` and `DataStore`, like other BPMN standard IO mappings, are supported by Camunda for modeling purposes only."? We could leave out "Camunda does not support any additional data features the BPMN standard defines."

Comment (christinaausley) [reply docs/components/modeler/bpmn/bpmn-coverage.md:144]: Whoops I think we forgot about this @nikku -- WDYT here? CC @aleksander-dytko

Comment (nikku) [reply docs/components/modeler/bpmn/bpmn-coverage.md:144]: > We could leave out "Camunda does not support any additional data features the BPMN standard defines."

Yea, we could leave it out, or link to our own support for I/O. Probably easiest to leave it out.

**Comment by christinaausley** [comment]
@nikku This is ready for final review 👍

**Comment by christinaausley** [comment]
I am going to go ahead and merge this. If needed, can backport in a separate PR. CC @aleksander-dytko as I know he is OOO.

**Comment by aleksander-dytko** [comment]
@christinaausley thanks for following up! 

This feature was introduced [with 8.2](https://github.com/camunda/product-hub/issues/618) - please backport this feature for versions <=8.2. Thanks!
